zerver/tests/test_message_delete.py

```diff
@@ -689,6 +701,45 @@ def check_delete_message_by_sender(sender_name: str, error_msg: str | None = Non
         check_delete_message_by_sender("cordelia")
         check_delete_message_by_sender("polonius")
 
+        do_change_realm_permission_group_setting(
+            realm, "can_delete_own_message_group", nobody_system_group, acting_user=None
+        )
+
+        do_change_stream_group_based_setting(
+            stream,
+            "can_delete_own_message_group",
+            members_system_group,
+            acting_user=iago,
+        )
+        # Users in per-channel `can_delete_own_message_group` can delete their
+        # own messages.
+        check_delete_message_by_sender("hamlet")
+        check_delete_message_by_sender(
+            "polonius", "You don't have permission to delete this message"
+        )
+
+        message_id = self.send_personal_message(hamlet, iago)
+        self.login("hamlet")
+        result = self.client_delete(f"/json/messages/{message_id}")
+        self.assert_json_error(result, "You don't have permission to delete this message")
```

Comment:
I think this can be tested at the end of this test. Also, I don't see a similar DM test for teh `can_delete_any_message_group` setting, would be good to add one for that setting as well.